src/plugins/intel_npu/src/plugin/npuw/compiled_model.hpp

```diff
@@ -109,6 +112,9 @@ class CompiledModel : public ov::npuw::ICompiledModel {
     std::string global_mem_device() const;
     std::string funcall_mem_device(const std::size_t idx) const;
 
+    //
+    void set_infer_request_listener(std::weak_ptr<ov::npuw::IInferRequestSubmissionListener> lst);
+    std::weak_ptr<ov::npuw::IInferRequestSubmissionListener> m_listener;
```

Comment:
doesn't look right for the compiled model

Reply:
but it is correct way - compiled model used as factory by LLMInrefRequest , since we need to create inferrequest of certain type, the only way i see is to inject listener in factory itself. that can be discussed more - but anyway LLMInferRequest cannot control composite infer-request like just or unfold,